### PR TITLE
Switch beta fixes, binary mapper support, resnet alternative

### PR DESCRIPTION
**Pull request by ddidacus**
hello everyone,
here I report some fixes hoping to get a better switching unit:

- added support for the binary mapper in the discovery phase
- fixed switch betas processing for visualization and entropy triggering (it needed masking)
- added feasibility restoration when the negative entorpy loss kicks in (temporarily deactivating obs and action recon losses)
- added support for symbolic observations (no resnet, no RGB) in babyAI miniboss
- also, I disabled the use of `train_discovery()` as I noticed the `requires_grad` attributes of the module to disable were not set to `False`! So I manually enforced it.

let me know what you end up seeing on your side!

### CODE REVIEW

**Comment by lucidrains** [comment]
lgtm otherwise!
